src/main/java/org/hydev/mcpm/client/database/searchusecase/SearcherByKeyword.java

```diff
@@ -0,0 +1,77 @@
+package org.hydev.mcpm.client.database.searchusecase;
+
+import org.hydev.mcpm.client.models.PluginModel;
+
+import java.util.*;
+
+/**
+ * Searcher that returns a map based on keywords.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearcherByKeyword implements Searcher {
+
+    private static Map<String, List<PluginModel>> keywordMap = null;
+
+    /**
+     * Returns a dictionary mapping the different keywords to the matching plugins
+     * This function should ignore letter case in searching.
+     * For example, searching "java" would match "Java" as well.
+     * This function implements fuzzy search that doesn't require the exact phrase to be available,
+     * but requires all words in the phrase to be present.
+     * For example, searching "java 11" would match "java jdk 11" but not "java"
+     *
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating each keyword to the matching plugins.
+     */
+    @Override
+    public Map<String, List<PluginModel>> constructSearchMaps(List<PluginModel> plugins) {
+        Map<String, List<PluginModel>> models = new HashMap<>();
+        for (PluginModel plugin : plugins) {
+            // Get latest version
+            var v = plugin.getLatestPluginVersion();
+            if (v.isPresent() && v.get().meta() != null &&
+                    v.get().meta().description() != null && !v.get().meta().description().equals("")) {
+                String[] keywords = v.get().meta().description()
+                        .replaceAll("[^a-zA-Z ]", "").toLowerCase().split("\\s+");
+                for (String keyword : keywords) {
+                    if (!models.containsKey(keyword))
+                        models.put(keyword, new ArrayList<>());
+                    models.get(keyword).add(plugin);
```

Comment:
It might save some work to use `getOrDefault` here...
```suggestion
                    var keywordList = models.getOrDefault(keyword, null);
                    if (keywordList == null) {
                        keywordList = new ArrayList<>();
                        models.put(keyword, keywordList);
                    }
                    keywordList.add(plugin);
```

Reply:
i prefer to keep what i had since it's shorter

Reply:
Alright, just figure that you're doubling the work (searching the array twice, once in `containsKey` and once again in `get`).

Reply:
ah true getOrDefault is cleaner. ill do that

Reply:
its just a bit quicker lol